queries.sql

```diff
@@ -8,3 +8,61 @@ SELECT name,escape_attempt FROM animals WHERE weight_kg>10.5;
 SELECT * FROM animals WHERE neutered=true;
 SELECT * FROM animals WHERE name<>'Gabumon';
 SELECT * FROM animals WHERE weight_kg>=10.4 AND weight_kg<=17.3;
+
+-- First transaction block of codes
+BEGIN;
+UPDATE animals 
+SET species = 'unspecified';
+SELECT species from animals; 
+ROLLBACK;
+SELECT species from animals;
+
+
+-- Second Transaction block of codes
+BEGIN;
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species ='pokemon' WHERE species IS NULL;
+SELECT species FROM animals;
+COMMIT;
+SELECT species FROM animals;
+
+-- Third Transaction block of codes
+BEGIN;
+DELETE FROM animals;
+SELECT COUNT(*) FROM animals;
+ROLLBACK;
+SELECT COUNT(*) FROM animals;
+COMMIT;
+
+-- Fourth Transaction block of codes
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals SET weight_kg=(weight_kg*-1);
+ROLLBACK TO SP1;
+UPDATE animals SET weight_kg=(weight_kg*-1) WHERE weight_kg<0;
+COMMIT;
+
+-- how many animals are there?
+SELECT COUNT(*) AS total_animals FROM animals;
+
+-- how many animals have never tried to escape
+SELECT COUNT(*) AS no_escape FROM animals WHERE escape_attempt=0;
+
+-- what is the average weight on animals
+SELECT AVG(weight_kg) AS average_weight FROM animals;
```

Comment:
- [ ] You haven't updated the image screenshot of the changes asked by the previous reviewer to have the value as `15.55`. kindly update the screenshot